### PR TITLE
REM-530

### PR DESCRIPTION
**Pull request by imtithalhalalo**
adding combo group for purpose of exchange
page: 'ct-purpose-exchanges'

### CODE REVIEW

File: src/pages/ct-purpose-exchanges/forms/PurposeOfExchangeForm.js [hunk 1]
```diff
@@ -109,6 +110,23 @@ export default function PurposeOfExchangeForm({ labels, maxAccess, recordId, set
...
             </Grid>
+            <Grid item xs={12}>
+              <ResourceComboBox
+                endpointId={CurrencyTradingSettingsRepository.PurposeExchangeGroup.qry}
+                name='groupId'
+                label={labels.group}
+                valueField='recordId'
+                displayField='name'
+                values={formik.values}
+                required
```
Comment (RebeccaFrem02) [inline src/pages/ct-purpose-exchanges/forms/PurposeOfExchangeForm.js:121]: if the groupId is required add it to the validationSchema in the formik

Comment (imtithalhalalo) [reply src/pages/ct-purpose-exchanges/forms/PurposeOfExchangeForm.js:121]: not required

Comment (imtithalhalalo) [reply src/pages/ct-purpose-exchanges/forms/PurposeOfExchangeForm.js]: pusehed

**Comment by omarZrayka23** [comment]
recordId instead  recordId: recordId ? recordId : null,

**Comment by omarZrayka23** [comment]
Screenshot 2024-08-29 at 2 23 00 PM

in edit Mode second tab, i  clicked on  clear button then i clicked save t when closed window then opened i see data was not changed

File: src/pages/ct-purpose-exchanges/forms/PurposeOfExchangeForm.js [hunk 2]
```diff
@@ -25,14 +28,15 @@ export default function PurposeOfExchangeForm({ labels, maxAccess, recordId, set
...
     enableReinitialize: true,
     validateOnChange: true,
     validationSchema: yup.object({
-      name: yup.string().required(' '),
-      reference: yup.string().required(' ')
+      name: yup.string().required(),
+      reference: yup.string().required()
     }),
     onSubmit: async obj => {
       const recordId = obj.recordId
```
Comment (omarZrayka23) [inline src/pages/ct-purpose-exchanges/forms/PurposeOfExchangeForm.js:42]: remove it and put in if(obj.recordId) direct

**Comment by omarZrayka23** [comment]
in purpose wimdow  remove props not used ( height)

**Comment by omarZrayka23** [comment]
can u remove props are not used in InterfacesForm ?

**Review by omarZrayka23 — APPROVED** [review]

**Pull request merged**